Jenkinsfile

```diff
@@ -30,6 +30,17 @@ pipeline {
       }
     }
 
+    stage('Acceptance Tests') {
+      agent none
```

Comment:
What do these declarations within the stages do? 

Reading [some of the Jenkins documentation](https://jenkins.io/doc/book/pipeline/syntax/#agent), it looks like `agent none` is only for the top-level block - but I'm not finding their docs particularly clear.

Reply:
https://jenkins.io/doc/book/pipeline/jenkinsfile/#using-multiple-agents

The pipelines are set up to not use an agent as a whole now. This allows a stage which is only running meta tasks (waiting for user input, launching another job) to not consume an executor.

Reply:
Gotcha. Wonder if it works without the explicit declarations at the stage level (if the stages 'inherit' from the pipeline declaration). Happy to press on anyway.